File: 12356.py
def find_lr():
    temp_l,temp_r = 0,0    
    for i in range(l):        
        if soldiers[i] != 0:
            temp_l = soldiers[i]
    for i in range(r,s):
        if soldiers[i] != 0:
            temp_r = soldiers[i]
            break
    if temp_l == 0:
        temp_l = '*'
    if temp_r == 0:
        temp_r = '*'
    return temp_l,temp_r

s,b = list(map(int,input().split()))

while s != 0 and b != 0:        
    soldiers = [i for i in range(1,s+1)]    
    for i in range(b):
        soldiers_lr = []
        l,r = list(map(int,input().split()))        
        for j in range(l-1,r):
            soldiers[j] = 0        
        soldier_l,soldier_r = find_lr()                
        print(soldier_l,soldier_r)                    
    s,b = list(map(int,input().split()))
    print('-')

File: 12356.cpp
#include <iostream>
using namespace std;

int main() {
    int s,b,l,r; cin >> s >> b;    
    while(s != 0 && b != 0){
        int soldiers[s] = {0};
        for(int i = 0; i < b; i++){
            cin >> l >> r;
            for(int j = l-1; j < r; j++ ){
                soldiers[j] = s+1;
            }            
            int temp_l = s+1;
            int temp_r = s+1;            
            for(int j = l-2; j >= 0; j--){
                if(soldiers[j] != s+1){temp_l = j+1; break;}
            }
            for(int j = l-1; j < s; j++){
                if(soldiers[j] != s+1){
                    temp_r = j+1;
                    break;
                }
            }
            if(temp_l == s+1 && temp_r == s+1){cout<<"* *"<<endl;}
            else{
                if(temp_l == s+1){cout<<"* "<<temp_r<<endl;}
                if(temp_r == s+1){cout<<temp_l<<" *"<<endl;}
                if(temp_r != s+1 && temp_l != s+1){cout<<temp_l<<" "<<temp_r<<endl;}
            }
        }        
        cin >> s >> b;
        cout<<"-"<<endl;
    }
    return 0;
}